Use default parameters where possible in examples

jakobj opened the pull request:
As the title suggests. This should make all examples easier to understand.

fixes #346

HenrikMettler commented on examples/example_hurdles.py:
```diff
@@ -86,62 +81,34 @@ def objective_two(individual):
+# Next, we set up the evolutionary search. We first define the parameters of the
+# evolutionary algorithm. We define the upper percentile of individuals which
+# are evaluated on the (n+1)th objective by a list of numbers between 0 and 1
+ea_params = {"hurdle_percentile": [0.75, 0.0]}
```
Just out of curiosity: Why the percentile change?

jakobj replied:
just to make the runtime difference a bit more obvious. i thought i'd sneak it in with this PR ;)

HenrikMettler commented on examples/example_hurdles.py:
```diff
@@ -193,4 +161,4 @@ def recording_callback(pop):
 ax_function.set_xlabel(r"$x$")
 
-fig.savefig("example_minimal.pdf", dpi=300)
+fig.savefig("example_hurdles.pdf", dpi=300)
```
nice catch :)

jakobj replied:
yeah, we do like copy and paste :D

HenrikMettler commented on examples/example_parametrized_nodes.py:
```diff
@@ -96,23 +96,18 @@ def objective(individual, seed):
 # that we add the custom node defined above as a primitive.
 
-population_params = {"n_parents": 1, "seed": 818821}
+seed = 818821
```
`seed = 1234`

jakobj replied:
you mean bc it looks too hand picked? ;) should we change all seeds in the examples to 1234 then?

in general i think we can not avoid some handpicking just bc we want the examples to finish in reasonable time. a bad seed can prevent the search from finding a good solution. :(

HenrikMettler replied:
I just copied it from above, where you changed It in an example, however if possible we can just use the same everywhere?

jakobj replied:
yes, it seems like we can use the same everywhere. just pushed a new commit where the seeds are all identical (except a few cases where we want to use multiple different seeds in the same script).

HenrikMettler commented on examples/example_piecewise_target_function.py:
```diff
@@ -72,55 +72,35 @@ def objective(individual, rng):
-}
+# the genomes of individuals, and the evolutionary algorithm.
+
+seed = 818821
```
`seed = 1234`

HenrikMettler commented on examples/example_reorder.py:
```diff
@@ -50,59 +50,37 @@ def objective(individual):
+# the genome of individuals, and two evolutionary algorithms without (default)
+# and with genome reordering.
+
+seed = 818821
```
`seed = 1234`

HenrikMettler reviewed (changes requested):
Thanks a lot for taking care of this tedious task! I made some small observations. Also I didn't run the examples, I assume you did, to make sure they still work?

jakobj commented:
> Thanks a lot for taking care of this tedious task! I made some small observations. Also I didn't run the examples, I assume you did, to make sure they still work?

thanks for the review! addressed all your comments. i think i tried all examples, but in any case, CI should catch those that don't work. :crossed_fingers:

HenrikMettler reviewed (approved):
LG! Thanks for taking care of this! 🚀

HenrikMettler commented:
ah, before I merge, could you squash the two commits? thx

jakobj commented:
thx for the review! squashed and pushed, let's see whether CI is happy.
